netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -801,94 +852,239 @@ _net_ebpf_extension_connection_context_initialize(
+    }
+
     ExReleaseSpinLockExclusive(&_net_ebpf_ext_sock_addr_lock, old_irql);
 }
 
-static void
-_net_ebpf_ext_insert_connection_context_to_list(_Inout_ net_ebpf_extension_connection_context_t* connection_context)
+_Requires_exclusive_lock_held_(_net_ebpf_ext_sock_addr_lock) static ebpf_result_t
+    _net_ebpf_ext_insert_connection_context_to_low_memory_list(
+        _In_ const bpf_sock_addr_t* sock_addr_ctx, UINT64 transport_endpoint_handle)
 {
-    KIRQL old_irql = ExAcquireSpinLockExclusive(&_net_ebpf_ext_sock_addr_lock);
+    ebpf_result_t result = EBPF_SUCCESS;
+    PLIST_ENTRY entry = NULL;
+    net_ebpf_extension_connection_context_t* blocked_connection_context = NULL;
+    if (IsListEmpty(&_net_ebpf_ext_low_memory_free_connect_context_list)) {
+        NET_EBPF_EXT_BAIL_ON_ERROR_RESULT(EBPF_NO_MEMORY);
+    }
 
-    // Insert the most recent entry at the head.
-    InsertHeadList(&_net_ebpf_ext_connect_context_list, &connection_context->list_entry);
-    _net_ebpf_ext_connect_context_count++;
+    // Retrieve an entry from the pre-allocated list
+    entry = RemoveHeadList(&_net_ebpf_ext_low_memory_free_connect_context_list);
+    blocked_connection_context = CONTAINING_RECORD(entry, net_ebpf_extension_connection_context_t, list_entry);
+    memset(blocked_connection_context, 0, sizeof(net_ebpf_extension_connection_context_t));
 
-    // Purge stale entries from the list.
-    _net_ebpf_ext_purge_lru_contexts_under_lock(FALSE);
+    _net_ebpf_extension_connection_context_initialize(
+        sock_addr_ctx,
+        transport_endpoint_handle,
+        CONNECTION_CONTEXT_INITIALIZATION_SET_TIMESTAMP,
+        blocked_connection_context);
+
+    // Insert into the blocked context list
+    InsertHeadList(&_net_ebpf_ext_low_memory_blocked_connect_context_list, &blocked_connection_context->list_entry);
+
+Exit:
+    return result;
+}
+
+static ebpf_result_t
+_net_ebpf_ext_insert_connection_context_to_list(
+    _In_ const bpf_sock_addr_t* sock_addr_ctx, UINT64 transport_endpoint_handle)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    bool locked = false;
+    KIRQL old_irql = PASSIVE_LEVEL;
+    net_ebpf_extension_connection_context_t* blocked_connection_context =
+        (net_ebpf_extension_connection_context_t*)ExAllocatePoolUninitialized(
+            NonPagedPoolNx, sizeof(net_ebpf_extension_connection_context_t), NET_EBPF_EXTENSION_POOL_TAG);
 
+    NET_EBPF_EXT_BAIL_ON_ALLOC_FAILURE_RESULT(
+        NET_EBPF_EXT_TRACELOG_KEYWORD_SOCK_ADDR, blocked_connection_context, "blocked_connection", result);
+    memset(blocked_connection_context, 0, sizeof(net_ebpf_extension_connection_context_t));
+
+    _net_ebpf_extension_connection_context_initialize(
+        sock_addr_ctx,
+        transport_endpoint_handle,
+        CONNECTION_CONTEXT_INITIALIZATION_SET_TIMESTAMP,
+        blocked_connection_context);
+
+    old_irql = ExAcquireSpinLockExclusive(&_net_ebpf_ext_sock_addr_lock);
+    locked = true;
+
+    // Insert into hash table
```

Comment:
```suggestion
    // Insert into hash table.
```
And look for any other inconsistent lines in this file.